File: hamersapp/src/main/java/nl/ecci/Hamers/Helpers/SingleImageActivity.java
package nl.ecci.Hamers.Helpers;

import android.graphics.Bitmap;
import android.os.Bundle;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;
import android.widget.ImageView;
import com.nostra13.universalimageloader.core.DisplayImageOptions;
import com.nostra13.universalimageloader.core.ImageLoader;
import com.nostra13.universalimageloader.core.assist.FailReason;
import com.nostra13.universalimageloader.core.listener.ImageLoadingListener;
import nl.ecci.Hamers.R;
import uk.co.senab.photoview.PhotoViewAttacher;

public class SingleImageActivity extends AppCompatActivity {

    public static final String BEER_NAME = "BEER_NAME";
    public static final String IMAGE_URL = "IMAGE_URL";
    private String name;
    private String url;
    PhotoViewAttacher mAttacher;

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.single_image_activity);

        final Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        final ActionBar actionBar = getSupportActionBar();

        ImageView imageView = (ImageView) findViewById(R.id.single_image);

        name = getIntent().getStringExtra(BEER_NAME);
        url = getIntent().getStringExtra(IMAGE_URL);

        // Universal Image Loader
        ImageLoader imageLoader = ImageLoader.getInstance();
        DisplayImageOptions options = new DisplayImageOptions.Builder()
                .cacheInMemory(true)
                .cacheOnDisk(true)
                .considerExifParams(true)
                .bitmapConfig(Bitmap.Config.RGB_565)
                .build();

        imageLoader.displayImage(url, imageView, options, new ImageLoadingListener() {

            @Override
            public void onLoadingStarted(String imageUri, View view) {

            }

            @Override
            public void onLoadingFailed(String imageUri, View view, FailReason failReason) {

            }

            @Override
            public void onLoadingComplete(String imageUri, View view, Bitmap loadedImage) {
                mAttacher.update();
            }

            @Override
            public void onLoadingCancelled(String imageUri, View view) {

            }
        });
        mAttacher = new PhotoViewAttacher(imageView);

        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setHomeButtonEnabled(true);
            actionBar.setTitle(name);
        }


    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                this.finish();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: hamersapp/src/main/java/nl/ecci/hamers/helpers/SingleImageActivity.kt
package nl.ecci.hamers.helpers

import android.graphics.Bitmap
import android.os.Bundle
import android.support.v7.widget.Toolbar
import android.view.View
import com.google.gson.GsonBuilder
import com.nostra13.universalimageloader.core.ImageLoader
import com.nostra13.universalimageloader.core.assist.FailReason
import com.nostra13.universalimageloader.core.listener.ImageLoadingListener
import kotlinx.android.synthetic.main.activity_single_image.*
import nl.ecci.hamers.R
import nl.ecci.hamers.beers.Beer
import uk.co.senab.photoview.PhotoViewAttacher

class SingleImageActivity : HamersActivity() {

    private var mAttacher: PhotoViewAttacher? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_single_image)

        setSupportActionBar(toolbar as Toolbar?)

        val beer = GsonBuilder().create().fromJson(intent.getStringExtra(Beer.BEER), Beer::class.java)

        ImageLoader.getInstance().displayImage(beer.imageURL, beer_image, object : ImageLoadingListener {

            override fun onLoadingStarted(imageUri: String, view: View) {

            }

            override fun onLoadingFailed(imageUri: String, view: View, failReason: FailReason) {

            }

            override fun onLoadingComplete(imageUri: String, view: View, loadedImage: Bitmap) {
                mAttacher?.update()
            }

            override fun onLoadingCancelled(imageUri: String, view: View) {

            }
        })
        mAttacher = PhotoViewAttacher(beer_image)

        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setHomeButtonEnabled(true)
        supportActionBar?.title = beer.name
    }
}
